### ISSUE TITLE
when train and eval SSD or YOLO , presets img size

### ISSUE THREAD

**Issue by smartwell** [dmlc/gluon-cv#849]
First, thanks for your work.
At presets/yolo.py 171-174
```
h, w, _ = img.shape
interp = np.random.randint(0, 5)
img = timage.imresize(img, self._width, self._height, interp=interp)
bbox = tbbox.resize(bbox, (w, h), (self._width, self._height))
```
At presets/ssd.py 157-161
```
resize with random interpolation
h, w, _ = img.shape
interp = np.random.randint(0, 5)
img = timage.imresize(img, self._width, self._height, interp=interp)
bbox = tbbox.resize(bbox, (w, h), (self._width, self._height))
```
you want to resize img to required size, why not use timage.resize_contain(), train and eval use same method， i know other code such as pytorch-yolo or tf-yolo use latter way, i don't know what is effect about the two different ways.

Looking forward to reply.
@zhreshold

**Comment by zhreshold** [comment]
To batch images, they must be resized to same size. `resize_contain` is for arbitrary image shapes such as in the case of Faster/Mask RCNN

**Comment by smartwell** [comment]
Thanks for your reply. i know batch images must be same size, `resize_contain` also can make same size batch images, i means `resize_contain` can `keeping aspect ratio.` , i know `pytorch or tf  ` when train, they  keep aspect ratio with the same size images.

 i don't know what is effect about the two different ways.

> To batch images, they must be resized to same size. `resize_contain` is for arbitrary image shapes such as in the case of Faster/Mask RCNN

**Comment by zhreshold** [comment]
@smartwell I am confused. How do you resize the images to same resolution while keep aspect ratio while natural images have different apsect ratios?

**Comment by smartwell** [comment]
> @smartwell I am confused. How do you resize the images to same resolution while keep aspect ratio while natural images have different apsect ratios?

At [gluoncv/data/transforms/presets/ssd.py 163-166](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/data/transforms/presets/ssd.py)

i change code to :
```
        h, w, _ = img.shape
        interp = np.random.randint(0, 5)

        # img = timage.imresize(img, self._width, self._height, interp=interp)
        # bboxes = tbbox.resize(bboxes, (w, h), (self._width, self._height))

        img,  (off_x, off_y, scaled_x, scaled_y) = resize_contain(img, (self._width, self._height,), interp=interp)
        bboxes = tbbox.resize(bboxes, (w, h), (scaled_x, scaled_y))
        bboxes = tbbox.translate(bboxes, off_x, off_y)
```

**Comment by zhreshold** [comment]
@smartwell Does it work for batch_size > 1?

**Comment by smartwell** [comment]
> @smartwell Does it work for batch_size > 1?

yes，it works. just change little` resize_contain` ,add `interp=interp`
```
INFO:root:[Epoch 0][Batch 99], Speed: 50.085 samples/sec, CrossEntropy=28.117, SmoothL1=9.086
INFO:root:[Epoch 0][Batch 199], Speed: 48.340 samples/sec, CrossEntropy=16.579, SmoothL1=6.086
```
